serving/resource_model_serving.go

```diff
@@ -0,0 +1,75 @@
+package serving
+
+import (
+	"context"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/retries"
+	"github.com/databricks/databricks-sdk-go/service/endpoints"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const DefaultProvisionTimeout = 45 * time.Minute
+
+func ResourceModelServing() *schema.Resource {
+	s := common.StructToSchema(
+		endpoints.CreateServingEndpoint{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			m["name"].ForceNew = true
+			return m
+		})
+
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var e endpoints.CreateServingEndpoint
+			common.DataToStructPointer(d, s, &e)
+			endpoint, err := w.ServingEndpoints.CreateAndWait(ctx, e, retries.Timeout[endpoints.ServingEndpointDetailed](DefaultProvisionTimeout))
+			if err != nil {
+				return err
+			}
+			d.SetId(endpoint.Name)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			endpoint, err := w.ServingEndpoints.GetByName(ctx, d.Id())
+			if err != nil {
+				return err
+			}
+			return common.StructToData(*endpoint, s, d)
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var e endpoints.CreateServingEndpoint
+			common.DataToStructPointer(d, s, &e)
+			e.Config.Name = e.Name
+			_, err = w.ServingEndpoints.UpdateConfigAndWait(ctx, e.Config, retries.Timeout[endpoints.ServingEndpointDetailed](DefaultProvisionTimeout))
+			if err != nil {
+				return err
+			}
+			return nil
+		},
```

Comment:
I think I had a question along the lines of:

* How do I know what API fields are covered by the terraform resource? Is that captured explicitly somewhere?
*  How do we deal with PENDING states - is it enough to use UpdateConfigAndWait() or do we need to do this with separate calls? E.g., initialize the update in one call, then use a separate RPC to keep pinging to get status and only return once the status is READY (not PENDING).

Reply:
Yeah the doc file covers the arguments that are covered by the resource, and there are no new attributes that we are exposing in state, so it should only be those API fields.

UpdateConfigAndWait() and the create-equivalent poll for us and take care of the pending logic

Reply:
On the first question - sorry for being obtuse - I meant, where do we set fields that are read-only and not update-able via terraform. How do we control that? Is it somewhere in the Open API spec?

On the second question, I see, that's awesome!
https://src.dev.databricks.com/databricks/databricks-sdk-go/-/blob/service/endpoints/api.go?L218-254

Reply:
Yeah actually we do that by setting the overall schema to `endpoints.CreateServingEndpoint` so all the fields stored in state would be only those specified in the [struct](https://src.dev.databricks.com/databricks/databricks-sdk-go/-/blob/service/endpoints/model.go?L25). And as you can see, the only fields in this schema are the ones we care about and are editable, so we don't expose the rest.